exercises/chapter7/src/Data/AddressBook/Validation.purs

```diff
@@ -10,74 +10,129 @@ import Data.Traversable (traverse)
-    <*> (nonEmpty "City" a.city *> pure a.city)
-    <*> (lengthIs "State" 2 a.state *> pure a.state)
+  address <$> (nonEmpty "Street"  a.street *> pure a.street)
+          <*> (nonEmpty "City"    a.city   *> pure a.city)
+          <*> (lengthIs "State" 2 a.state  *> pure a.state)
+-- ANCHOR_END: validateAddress
 
+-- ANCHOR: validateAddressAdo
 validateAddressAdo :: Address -> V Errors Address
 validateAddressAdo a = ado
-  street <- (nonEmpty "Street" a.street *> pure a.street)
-  city <- (nonEmpty "City" a.city *> pure a.city)
-  state <- (lengthIs "State" 2 a.state *> pure a.state)
+  street <- (nonEmpty "Street"  a.street *> pure a.street)
+  city   <- (nonEmpty "City"    a.city   *> pure a.city)
+  state  <- (lengthIs "State" 2 a.state  *> pure a.state)
   in address street city state
+-- ANCHOR_END: validateAddressAdo
 
+-- ANCHOR: validatePhoneNumber
 validatePhoneNumber :: PhoneNumber -> V Errors PhoneNumber
 validatePhoneNumber pn =
   phoneNumber <$> pure pn."type"
-    <*> (matches "Number" phoneNumberRegex pn.number *> pure pn.number)
+              <*> (matches "Number" phoneNumberRegex pn.number *> pure pn.number)
+-- ANCHOR_END: validatePhoneNumber
 
+-- ANCHOR: validatePhoneNumberAdo
 validatePhoneNumberAdo :: PhoneNumber -> V Errors PhoneNumber
 validatePhoneNumberAdo pn = ado
-  tpe <- pure pn."type"
+  tpe    <- pure pn."type"
   number <- (matches "Number" phoneNumberRegex pn.number *> pure pn.number)
   in phoneNumber tpe number
+-- ANCHOR_END: validatePhoneNumberAdo
 
+-- ANCHOR: validatePerson
 validatePerson :: Person -> V Errors Person
 validatePerson p =
   person <$> (nonEmpty "First Name" p.firstName *> pure p.firstName)
-    <*> (nonEmpty "Last Name" p.lastName *> pure p.lastName)
-    <*> validateAddress p.homeAddress
-    <*> (arrayNonEmpty "Phone Numbers" p.phones *> traverse validatePhoneNumber p.phones)
+         <*> (nonEmpty "Last Name" p.lastName   *> pure p.lastName)
+         <*> validateAddress p.homeAddress
+         <*> (arrayNonEmpty "Phone Numbers" p.phones *>
+              traverse validatePhoneNumber p.phones)
+-- ANCHOR_END: validatePerson
 
+-- ANCHOR: validatePersonAdo
 validatePersonAdo :: Person -> V Errors Person
 validatePersonAdo p = ado
   firstName <- (nonEmpty "First Name" p.firstName *> pure p.firstName)
-  lastName <- (nonEmpty "Last Name" p.lastName *> pure p.lastName)
-  address <- validateAddress p.homeAddress
-  numbers <- (arrayNonEmpty "Phone Numbers" p.phones *> traverse validatePhoneNumber p.phones)
+  lastName  <- (nonEmpty "Last Name" p.lastName   *> pure p.lastName)
+  address   <- validateAddress p.homeAddress
+  numbers   <- (arrayNonEmpty "Phone Numbers" p.phones *>
+                traverse validatePhoneNumber p.phones)
   in person firstName lastName address numbers
+-- ANCHOR_END: validatePersonAdo
 
+-- ANCHOR: validatePerson
 validatePerson' :: Person -> Either Errors Person
 validatePerson' p = unV Left Right $ validatePerson p
+-- ANCHOR_END: validatePerson
```

Comment:
Collision follow-up part-2. Example of prime suffix taken.

My vim anchor-creating macro mishandled this by not picking-up the `'`. Not sure if that's a valid anchor name anyway.

It turns out that this snippet isn't even cited in the text. So **not sure if we should just delete it, or add some more chapter content** about `unV`.